src/media/js/elements/select.js

```diff
@@ -0,0 +1,263 @@
+            }
+        })
+    });
+
+    var MktSelectElement = document.registerElement('mkt-select', {
+        prototype: Object.create(HTMLSelectElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    // Create a hidden <select> element.
+                    var select = document.createElement('select');
+                    select.style.display = 'none';
+                    copyAttrs(select, root);
+
+                    // Create <mkt-option>/<mkt-optgroup> using <option>s.
+                    var mktOptGroup = document.createElement('mkt-optgroup');
+                    var optGroup = root.querySelector('optgroup');
+                    var options = root.querySelectorAll('option');
+                    for (var i = 0; i < options.length; i++) {
+                        var option = options[i];
+
+                        var mktOption = document.createElement('mkt-option');
+                        copyAttrs(mktOption, option);
+                        mktOption.innerHTML = option.innerHTML;
+                        mktOption.option = option;
+                        mktOptGroup.appendChild(mktOption);
+                    }
+
+                    // Add the <optgroup>s.
+                    root.appendChild(mktOptGroup);
+                    mktOptGroup.optGroup = optGroup;
+                    select.appendChild(optGroup);
+
+                    // Add the hidden <select>
+                    root.appendChild(select);
+                    root.select = select;
+
+                    // Update indices (for colors).
+                    root.updateOptionIndices();
+
+                    // Align options with <mkt-selected-text>.
+                    if (root.alignOptions() > 0) {
+                        root.setAttribute('data-mkt-select--aligned', true);
+                    }
+
+                    // Set up click handlers.
+                    root.addEventListener('click', function(e) {
+                        root.toggle();
+                        if (e.target.tagName == 'MKT-OPTION') {
+                            // Change value if click on option.
+                            root.value = e.target.value;
+                        }
+                    });
+                }
+            },
+            select: {
+                // Actual <select> element to proxy to, steal its interface.
+                // Value set in the createdCallback.
+                value: null,
+                writable: true
+            },
+            toggle: {
+                // Toggles visibility of the <mkt-option>s.
+                value: function(bool) {
+                    if (!this.getAttribute('data-mkt-select--aligned')) {
+                        if (this.alignOptions() > 0) {
+                            this.setAttribute('data-mkt-select--aligned', true);
+                        }
+                    }
+                    if (bool) {
```

Comment:
I don't think you need this `if`, you can just pass `bool` in and if it is `undefined` it will toggle.